Doesn't build on NixOS

I am using `hyprscroller.url = "github:dawsers/hyprscroller";` as a flake input to my NixOS system configuration, and I've added `hyprscroller` as a Hyprland plugin with:

```
wayland.windowManager.hyprland.plugins = [ inputs.hyprscroller.packages.${pkgs.stdenv.hostPlatform.system}.hyprscroller ]
```

I am using an x86_64-based system. When I build my system using this configuration, I get the following error indicating some kind of internal issue in the hyprscroller build process (see below).

```
building the system configuration...
error: builder for '/nix/store/jjs4k2i2rwwl12z4mgjphvzzdhj1awi9-hyprscroller-date=2024-09-02_462376d.drv' failed with exit code 2;
       last 10 log lines:
       > /build/source/src/scroller.h:11:7: note:   because the following virtual functions are pure within 'ScrollerLayout':
       >    11 | class ScrollerLayout : public IHyprLayout {
       >       |       ^~~~~~~~~~~~~~
       > In file included from /build/source/src/scroller.h:1:
       > /nix/store/30rri4jr45y2bamsyivpp25prx0jx9dn-hyprland-0.41.2+date=2024-07-03_e894d5e-dev/include/hyprland/src/layout/IHyprLayout.hpp:113:18: note:     'virtual void IHyprLayout::fullscreenRequestForWindow(PHLWINDOW, eFullscreenMode, bool)'
       >   113 |     virtual void fullscreenRequestForWindow(PHLWINDOW, eFullscreenMode, bool) = 0;
       >       |                  ^~~~~~~~~~~~~~~~~~~~~~~~~~
       > make[2]: *** [CMakeFiles/hyprscroller.dir/build.make:76: CMakeFiles/hyprscroller.dir/src/main.cpp.o] Error 1
       > make[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/hyprscroller.dir/all] Error 2
       > make: *** [Makefile:91: all] Error 2
       For full logs, run 'nix log /nix/store/jjs4k2i2rwwl12z4mgjphvzzdhj1awi9-hyprscroller-date=2024-09-02_462376d.drv'.
error: 1 dependencies of derivation '/nix/store/2yr97ymsxcgmp3i7gxji2rvjpnp1bjyb-hm_hyprhyprland.conf.drv' failed to build
error: 1 dependencies of derivation '/nix/store/q3cv1a72lb4zgv04nsk379b0lqh4b4lk-activation-script.drv' failed to build
error: 1 dependencies of derivation '/nix/store/ah39hcn702d6hwk55sbzm7sj8jrxp7gj-home-manager-generation.drv' failed to build
```

Reproduced by `nix build github:dawsers/hyprscroller`.

```
error: builder for '/nix/store/jjs4k2i2rwwl12z4mgjphvzzdhj1awi9-hyprscroller-date=2024-09-02_462376d.drv' failed with exit code 2;
       last 25 log lines:
       >       |                          ^~~~~~~~~~~
       > /build/source/src/scroller.h:57:29: error: 'WORKSPACEID' has not been declared
       >    57 |     Row *getRowForWorkspace(WORKSPACEID workspace);
       >       |                             ^~~~~~~~~~~
       > In file included from /nix/store/zc0nsv23pakbafngjy32kvhfzb16as43-gcc-13.3.0/include/c++/13.3.0/bits/shared_ptr_base.h:59,
       >                  from /nix/store/zc0nsv23pakbafngjy32kvhfzb16as43-gcc-13.3.0/include/c++/13.3.0/bits/shared_ptr.h:53,
       >                  from /nix/store/zc0nsv23pakbafngjy32kvhfzb16as43-gcc-13.3.0/include/c++/13.3.0/chrono:49,
       >                  from /nix/store/30rri4jr45y2bamsyivpp25prx0jx9dn-hyprland-0.41.2+date=2024-07-03_e894d5e-dev/include/hyprlan
d/sr
c/debug/Log.hpp:6,
       >                  from /nix/store/30rri4jr45y2bamsyivpp25prx0jx9dn-hyprland-0.41.2+date=2024-07-03_e894d5e-dev/include/hyprlan
d/sr
c/config/ConfigManager.hpp:6,
       >                  from /build/source/src/main.cpp:1:
       > /nix/store/zc0nsv23pakbafngjy32kvhfzb16as43-gcc-13.3.0/include/c++/13.3.0/bits/unique_ptr.h: In instantiation of 'constexpr s
td::
__detail::__unique_ptr_t<_Tp> std::make_unique(_Args&& ...) [with _Tp = ScrollerLayout; _Args = {}; __detail::__unique_ptr_t<_Tp> = __
deta
il::__unique_ptr_t<ScrollerLayout>]':
       > /build/source/src/main.cpp:50:1:   required from here
       > /nix/store/zc0nsv23pakbafngjy32kvhfzb16as43-gcc-13.3.0/include/c++/13.3.0/bits/unique_ptr.h:1070:30: error: invalid new-expre
ssio
n of abstract class type 'ScrollerLayout'
       >  1070 |     { return unique_ptr<_Tp>(new _Tp(std::forward<_Args>(__args)...)); }
       >       |                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
       > /build/source/src/scroller.h:11:7: note:   because the following virtual functions are pure within 'ScrollerLayout':
       >    11 | class ScrollerLayout : public IHyprLayout {
       >       |       ^~~~~~~~~~~~~~
       > In file included from /build/source/src/scroller.h:1:
       > /nix/store/30rri4jr45y2bamsyivpp25prx0jx9dn-hyprland-0.41.2+date=2024-07-03_e894d5e-dev/include/hyprland/src/layout/IHyprLayo
ut.h
pp:113:18: note:     'virtual void IHyprLayout::fullscreenRequestForWindow(PHLWINDOW, eFullscreenMode, bool)'
       >   113 |     virtual void fullscreenRequestForWindow(PHLWINDOW, eFullscreenMode, bool) = 0;
       >       |                  ^~~~~~~~~~~~~~~~~~~~~~~~~~
       > make[2]: *** [CMakeFiles/hyprscroller.dir/build.make:76: CMakeFiles/hyprscroller.dir/src/main.cpp.o] Error 1
       > make[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/hyprscroller.dir/all] Error 2
       > make: *** [Makefile:91: all] Error 2
       For full logs, run 'nix log /nix/store/jjs4k2i2rwwl12z4mgjphvzzdhj1awi9-hyprscroller-date=2024-09-02_462376d.drv'.
```

Full trace: [hyprscroller.log](https://github.com/user-attachments/files/16840940/hyprscroller.log)

--- a/.gitignore
+++ b/.gitignore
@@ -3,3 +3,4 @@ Release/
 hyprscroller.so
 .cache/
 compile_commands.json
+result

